pam/internal/gdm/conversation.go

```diff
@@ -14,7 +14,10 @@ import (
 )
 
 var conversations atomic.Int32
-var challengeRegex = regexp.MustCompile(`"challenge"\s*:\s*"(?:[^"\\]|\\.)*"`)
+var secretRegex = regexp.MustCompile(`"secret"\s*:\s*"(?:[^"\\]|\\.)*"`)
+
+// TODO(UDENG-5844): Remove this once the auth data field has been renamed to "secret".
+var secretRegexOld = regexp.MustCompile(`"challenge"\s*:\s*"(?:[^"\\]|\\.)*"`)
```

Comment:
Maybe we can still keep the todo but use instead:

```suggestion
var secretRegex = regexp.MustCompile(`"(secret|challenge)"\s*:\s*"(?:[^"\\]|\\.)*"`)
```

And then replace all with `"$1":"**************"` ?

Reply:
Mmh I don't know, it makes both this change and the change we need to do in the future a bit more complex, which means we also need to spend more time testing manually that it still works as expected (because it's security critical that this works as expected). The benefit is only that it's a bit faster because we don't have to check two regexes, right? Since these checks are only done when debug mode is enabled, I don't think we need to care much about the performance, so I'm leaning towards the simpler change.

Reply:
Ok fair enough...